pkg/scheduler/objects/application.go

```diff
@@ -1308,14 +1334,3 @@ func (sa *Application) notifyRMAllocationAskReleased(rmID string, released []*Al
 	}
 	sa.rmEventHandler.HandleEvent(releaseEvent)
 }
-
-// Auto progress the application when it enters the Failing state if there is nothing to clean up.
-// Since this is called by the _locked_ state machine while processing an event we cannot call back
-// into the statemachine directly and we need a go routine to avoid a deadlock.
-func (sa *Application) failAppIfPossible() {
-	if resources.IsZero(sa.pending) && resources.IsZero(sa.allocatedResource) && resources.IsZero(sa.allocatedPlaceholder) {
-		// The event handling cannot fail
-		//nolint: errcheck
-		go sa.HandleApplicationEvent(FailApplication)
-	}
-}
```

Comment:
I think we will need this change, because we want to fail the apps if there is a partition removed. In this case we might have an application with no allocations and asks, so if we are not moving it from Failing to Failed, it will be stuck into Failing state.

Reply:
When the partition is removed there is no way to get to the applications any more. Everything is gone nothing is kept. It does not matter what happens in the core as long as everything that needs to be released is communicated to the shim.
If we want to keep history around for a removed shim than this might become important but at the moment _everything_ is deleted from memory when the partition is removed.

Reply:
That mens that Failing the application here: https://github.com/apache/incubator-yunikorn-core/blob/master/pkg/scheduler/partition_manager.go#L131 makes no sense?

Reply:
Not for the core side. However that event might trigger a state change which might trigger updates to be send out to the shim.
About 15 line later we call back into the context and remove the partition from the context. You cannot access it anymore. I think there needs to be a broader discussion about historic information tracking in the objects. The core does not do any of that at the moment which we really need to revisit. At that point it will become important **but** that is a larger change that needs to be implemented.